src/windows/wslaservice/inc/wslaservice.idl

```diff
@@ -102,12 +109,93 @@ struct _VIRTUAL_MACHINE_SETTINGS {
 } VIRTUAL_MACHINE_SETTINGS;
 
 
-typedef
-struct _WSLA_SESSION_SETTINGS {
+
+struct WSLA_SESSION_SETTINGS {
     LPCWSTR DisplayName;
-    // Details TBD.
-} WSLA_SESSION_SETTINGS;
+    LPCWSTR StoragePath;
+
+    // TODO: Termination callback, flags
+};
+
 
+struct WSLA_REGISTRY_AUTHENTICATION_INFORMATION
+{
+    int Dummy; // Dummy value for the .idl file to compile.
+    // TODO: Details TBD.
+};
+
+
+struct WSLA_IMAGE_INFORMATION
+{
+    LPWSTR Name;
+    LPWSTR Hash;
+    ULONGLONG Size;
+    ULONGLONG DownloadTimestamp;
+};
+
+struct WSLA_PROCESS_OPTIONS
+{
+    LPCSTR Executable;
+    [unique] LPCSTR CurrentDirectory;
+    [size_is(CommandLineCount)] LPCSTR* CommandLine;
+    ULONG CommandLineCount;
+    [unique, size_is(EnvironmentCount)] LPCSTR* Environment;
+    ULONG EnvironmentCount;
+    [unique, size_is(EnvironmentCount)] WSLA_PROCESS_FD *Fds;
+    int FdsCount;
+};
+
+struct WSLA_VOLUME
+{
+    LPCSTR WindowsPath;
+    LPCSTR LinuxPath;
+};
+
+struct WSLA_PORT_MAPPING
+{
+    USHORT WindowsPort;
+    USHORT ContainerPort;
+};
+
+struct WSLA_CONTAINER_OPTIONS
+{
+    LPCWSTR Image;
+    LPCWSTR Name;
+    struct WSLA_PROCESS_OPTIONS* InitProcessOptions;
+    [unique, size_is(VolumesCount)] struct WSLA_VOLUME* Volumes;
+    ULONG VolumesCount;
+    [unique, size_is(PortsCount)] struct WSLA_PORT_MAPPING* Ports;
+    ULONG PortsCount;
+    DWORD Flags; // GPU, Privileged, ...
+    // TODO: List specific GPU devices.
+    ULONGLONG ShmSize;
+};
```

Comment:
I assume we can add more options based on customer need? Like the ulimit nofile we discussed.